src/content/developers/docs/smart-contracts/formal-verification/index.md

```diff
@@ -0,0 +1,272 @@
+
+### Program verifiers for checking correctness {#program-verifiers}
+
+**Solidity SMTChecker** - *_Solidity’s SMTChecker is a built-in model checker based on SMT (Satisfiability Modulo Theories) and Horn solving. It confirms if a contract’s source code matches specifications during compilation and statically checks for violations of safety properties._*
+
+- [GitHub](https://github.com/ethereum/solidity)
+
+**solc-verify** - *_solc-verify is an extended version of the Solidity compiler that can perform automated formal verification on Solidity code using annotations and modular program verification._*
+
+- [GitHub](https://github.com/SRI-CSL/solidity)
+
+**hevm** - *_hevm is a symbolic execution engine and equivalence checker for EVM bytecode._*
+
+- [GitHub](https://github.com/dapphub/dapptools/tree/master/src/hevm)
+
+**KEVM**: _*KEVM is a formal semantics of the Ethereum Virtual Machine (EVM) written in the K framework. KEVM is executable and can prove certain property-related assertions using reachability logic.*_
+
+- [GitHub](https://github.com/runtimeverification/evm-semantics)
+- [Documentation](https://jellopaper.org/)
+
+### Symbolic execution-based tools for detecting vulnerable patterns in smart contracts {#symbolic-execution-tools}
+
+**Manticore** - *_A tool for analyzing EVM bytecode analysis tool based on symbolic execution_.*
+
+- [GitHub](https://github.com/trailofbits/manticore)
+- [Documentation](https://github.com/trailofbits/manticore/wiki)
+
+**Securify** - _*A security analyzer for analyzing Solidity source code in Ethereum smart contracts and detecting property violations.*_
+
+- [GitHub](https://github.com/eth-sri/securify2)
+
+**Oyente** - *_Static analysis tool for analyzing vulnerabilities in EVM bytecode with symbolic execution._*
+
+- [GitHub](https://github.com/melonproject/oyente)
+
+**Echidna** - *_Echidna is a fast contract fuzzer for detecting vulnerabilities in smart contracts through property-based testing._*
+
+- [GitHub](https://github.com/crytic/echidna/)
+
+**Harvey** - *_Harvey is an automated fuzzing tool useful for detecting property violations in smart contract code._*
+
+- [Website](https://consensys.net/diligence/fuzzing/)
+
+## Further reading {#further-reading}
+
+### Articles {#articles}
+
+- [How Formal Verification of Smart Contracts Works](https://runtimeverification.com/blog/how-formal-verification-of-smart-contracts-works/) [How Formal Verification Can Ensure Flawless Smart Contracts](https://media.consensys.net/how-formal-verification-can-ensure-flawless-smart-contracts-cbda8ad99bd1)
+- [An Overview of Formal Verification Projects in the Ethereum Ecosystem](https://github.com/leonardoalt/ethereum_formal_verification_overview)
+- [An Introduction to Formal Verification of Ethereum Contracts](https://github.com/pirapira/Ethereum-Formal-Verification-Overview)
+- [SMTChecker and Formal Verification](https://docs.soliditylang.org/en/v0.8.15/smtchecker.html)
+
```

Comment:
Maybe worth adding https://runtimeverification.com/blog/end-to-end-formal-verification-of-ethereum-2-0-deposit-smart-contract/ as well?

Reply:
Added this too.